bashflux/message.go

```diff
@@ -0,0 +1,55 @@
+package cmd
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+const contentTypeSenml = "application/senml+json"
+
+var cmdMessages = []cobra.Command{
+	cobra.Command{
+		Use:   "send",
+		Short: "send <channel_id> <JSON_string> <client_token>",
+		Long:  `Sends message on the channel`,
+		Run: func(cmdCobra *cobra.Command, args []string) {
+			if len(args) == 3 {
+				SendMsg(args[0], args[1], args[2])
+			} else {
+				LogUsage(cmdCobra.Short)
+			}
+		},
+	},
+}
+
+func NewMessagesCmd() *cobra.Command {
+	cmd := cobra.Command{
+		Use:   "msg",
+		Short: "Send or retrieve messages",
+		Long:  `Send or retrieve messages: control message flow on the channel`,
+	}
+
+	for i, _ := range cmdMessages {
```

Comment:
Use value instead of index.

Reply:
`cmdMessages` is not struct as far as I can tell: `var cmdMessages = []cobra.Command{`

Reply:
@manuIO Please explain to me why wouldn't it work. I have working example here: https://play.golang.org/p/NYWE0SZ84am.

Reply:
@manuIO For some reason my comment was not posted. Problem isn't in `cobra.Command`.  I missed the fact that you are using pointer here. Value is declared at the beginning of for statement, so it just changes values, but its memory address doesn't change, so when you try to use pointer to the value, it is referencing same address every time. There are two solutions for this. One is to copy value to new variable inside for loop, and the other one is what you've already did. I prefer you solution. What I would suggest is to remove `_` from `for` statement.

Reply:
@anovakovic01 is this one resolved?